boards.txt

```diff
@@ -228,7 +228,11 @@ Generic_nRF51822.build.board=GENERIC
 Generic_nRF51822.build.core=nRF5
 Generic_nRF51822.build.variant=Generic
 Generic_nRF51822.build.variant_system_lib=
+<<<<<<< Updated upstream
+Generic_nRF51822.build.extra_flags=-DNRF51 -DNRF51822
+=======
 Generic_nRF51822.build.extra_flags=-DNRF51
+>>>>>>> Stashed changes
```

Comment:
I think `-1` would be used for pins that "do not exists", but still need to be in the array for other pins to not take that value.
So if you have a header with 4 pins, 0 to 3, but pin 2 in the header is marked with something that is not a microcontroller GPIO (so maybe ground or Vcc), you still need a `g_ADigitalPinMap[]` with 4 values, so then it'd be `g_ADigitalPinMap[] = { 0, 1, -1, 50}`. It doesn't matter what the nordic pin values are (as long as they are real pins), so you can add them in whatever order makes sense for your board breakout, since that's the array that does the mapping between "arduino pin" and "nordic pin". So in the example above `digitalRead(3)` maps to pin 50 if the nRF.
Did that make sense? does that answer your question?